Data/Pool.hs

```diff
@@ -403,6 +437,16 @@ destroyResource Pool{..} LocalPool{..} resource = do
    atomically (modifyTVar_ inUse (subtract 1))
 {-# INLINABLE destroyResource #-}
 
+giveResource :: Pool a -> a -> IO ()
```

Comment:
nit: a one line comment about what the function does would be helpful

Reply:
Do you think a better name would be `releaseResource`? That is, it either frees it back to the pool or destroys it.